app/src/main/java/org/gnucash/android/importer/CommoditiesXmlHandler.java

```diff
@@ -82,6 +82,7 @@ public void startElement(String uri, String localName, String qName, Attributes
 
     @Override
     public void endDocument() throws SAXException {
+        mCommoditiesDbAdapter.deleteAllRecords();
```

Comment:
What is the purpose of this call? Aren't we always importing into a new book anyway? (which means a fresh database)

Reply:
no, an existing book that will have its database "upgraded" means that the table is already populated with commodities, so we need to delete the old ones and populate from the latest xml.